source/connection-troubleshooting.txt

```diff
@@ -0,0 +1,308 @@
+
+.. code-block:: none
+   :copyable: false
+
+   Error: couldn't connect to server 127.0.0.1:27017
+
+The following sections describe actions you can take to potentially resolve the
+issue.
+
+.. _csharp-troubleshooting-connection-string-port:
+
+Check Your Connection String
+----------------------------
+
+Verify that the hostname and port number in the connection string are both
+accurate. The default port value for a MongoDB instance is
+``27017``, but you can configure MongoDB to communicate on another port.
+
+.. _csharp-troubleshooting-connection-firewall:
+
+Configure Your Firewall
+-----------------------
+
+Verify that the ports your MongoDB deployment listens on are not blocked by a
+firewall on the same network. MongoDB uses port ``27017`` by default. To learn
+more about the default ports MongoDB uses and how to change them, see
+:manual:`Default MongoDB Port </reference/default-mongodb-port/>`.
+
+.. warning::
+
+   Do not open a port in your firewall unless you are sure it's the port
+   used by your MongoDB deployment.
+
+Authentication Error
+~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} can fail to connect to a MongoDB instance if
+the authorization is not configured correctly. If you are using ``SCRAM-SHA-256``
+or ``SCRAM-SHA-1`` for authentication and the driver fails to connect, the
+driver might raise an error message similar to one of the following messages:
+
+.. code-block:: none
+   :copyable: false
+
+   Command failed with error 18 (AuthenticationFailed): 'Authentication
+   failed.' on server <hostname>:<port>.
+
+.. code-block:: none
+   :copyable: false
+
+   Authentication failed","attr":{"mechanism":"SCRAM-SHA-256","principalName":
+   "<user>","<auth databse>":"<user>","client":"127.0.0.1:2012",
+   "result":"UserNotFound: Could not find user}}
+
+.. code-block:: none
+   :copyable: false
+
+   connection() error occurred during connection handshake: auth error:
+   sasl conversation error: unable to authenticate using mechanism
+   "SCRAM-SHA-256": (AuthenticationFailed) Authentication failed.
+
+The following sections describe actions you can take to potentially resolve the
+issue.
+
+.. _csharp-troubleshooting-connection-string-auth:
+
+Check Your Connection String
+----------------------------
+
+An invalid connection string is the most common cause of authentication
+issues when attempting to connect to MongoDB using connection strings and
+``SCRAM-SHA-256`` or ``SCRAM-SHA-1``.
+
+.. tip::
+
+   For more information about connection strings,
+   see :ref:`Connection URI <csharp-connection-uri>` in the Connection Guide.
+
+If your connection string contains a username and password, ensure that they
+are in the correct format. If the username or password includes any of the
+following characters, they must be
+`percent encoded <https://tools.ietf.org/html/rfc3986#section-2.1>`__:
+
+.. code-block:: none
+
+    : / ? # [ ] @
+
+The following example shows how to percent encode "#MyP@assword?":
```

Comment:
I: On my machine, #MyP@assword creates a link to draft an email for some reason. Since we don't want users to experience unexpected linking out of our docs, could you change this to something that doesn't get picked up as a link/action?